### PR TITLE
Update brokercell and channel CRDs API version

### PR DESCRIPTION
**Pull request by bharattkukreja**
Fixes https://github.com/google/knative-gcp/issues/1475

## Proposed Changes

- Update brokercell and channel CRDs to use apiextensions.k8s.io/v1 as the API version

### CODE REVIEW

**Comment by bharattkukreja** [comment]
cc @capri-xiyue

**Comment by capri-xiyue** [comment]
Besides changing the version, I think you also need to follow this PR https://github.com/google/knative-gcp/pull/1473/files#diff-03dcdd5a50d164008de2bb1c827d7cab to modify other things like adding `conversionReviewVersions: ["v1", "v1beta1"]`

**Comment by capri-xiyue** [comment]
In addition, it will be better if you can use yaml merge to rewrite the channel yaml

**Comment by capri-xiyue** [comment]
https://yaml.org/type/merge.html

File: config/core/resources/channel.yaml [hunk 1]
```diff
@@ -36,293 +36,189 @@ spec:
...
+            <<: *status
+            properties:
+              <<: *statusProperties
+              serviceAccountName:
+                type: string
+  - <<: *version
+    name: v1beta1
+    served: true
+    storage: true
+    # the schema of v1beta1 is exactly the same as v1alpha1 schema
```
Comment (capri-xiyue) [inline config/core/resources/channel.yaml:223]: @Harwayne Is schema of channel v1alpha1 the same as channel v1beta1?

File: config/core/resources/channel.yaml [hunk 2]
```diff
@@ -36,293 +36,189 @@ spec:
...
   scope: Namespaced
-  subresources:
-    status: {}
   preserveUnknownFields: false
   conversion:
     strategy: Webhook
-    webhookClientConfig:
+    # conversionReviewVersions indicates what ConversionReview versions are understood/preferred by the webhook.
+    # The first version in the list understood by the API server is sent to the webhook.
+    # The webhook must respond with a ConversionReview object in the same version it received.
```
Comment (capri-xiyue) [inline config/core/resources/channel.yaml:44]: I think you are missing something. The e2e failed with applying this yaml.  https://github.com/bharattkukreja/knative-gcp/blob/dee416986d74f3cf671ab2f7e25264b54c190c47/config/core/resources/cloudpubsubsource.yaml#L42

File: config/core/resources/channel.yaml [hunk 3]
```diff
@@ -36,293 +36,190 @@ spec:
...
+      # The first version in the list understood by the API server is sent to the webhook.
+      # The webhook must respond with a ConversionReview object in the same version it received.
+      conversionReviewVersions: ["v1", "v1beta1"]
+      clientConfig:
+        service:
+          name: webhook
+          namespace: cloud-run-events
+  versions:
+  - &version
+    name: v1
```
Comment (capri-xiyue) [inline config/core/resources/channel.yaml:53]: Channel doesn't have v1

**Comment by capri-xiyue** [comment]
I just found that yaml merge syntax can handle it elegantly with field removal or field addition or exact same field. But for field replacement like Bharatt did for channel in this PR, it looks like  the usage of `<<:` to inject the differences between versions doesn't reduce the needed efforts.
In addition, later when we want to deprecate older versions, the usage of  `<<:`  and `&` make things more complex when we have filed replacement scenario. It looks like in this case, CRD generation is still quite useful when we have kind of complex differences between versions. @Harwayne  What do you think of it?

**Comment by danyinggu** [comment]
> I just found that yaml merge syntax can handle it elegantly with field removal or field addition or exact same field. But for field replacement like Bharatt did for channel in this PR, it looks like the usage of `<<:` to inject the differences between versions doesn't reduce the efforts.
> In addition, later when we want to deprecate older versions, the usage of `<<:` and `&` make things more complex when we have filed replacement scenario. It looks like in this case, CRD generation is still quite useful when we have kind of complex differences between versions. @Harwayne What do you think of it?

I slightly checked his PR. I think for channel CRD, maybe he could put v1beta1 which have less field as the first schema, then insert additional fields that v1alpha1 have as the second schema, maybe this will make the syntax cleaner?

**Comment by bharattkukreja** [comment]
> > I just found that yaml merge syntax can handle it elegantly with field removal or field addition or exact same field. But for field replacement like Bharatt did for channel in this PR, it looks like the usage of `<<:` to inject the differences between versions doesn't reduce the efforts.
> > In addition, later when we want to deprecate older versions, the usage of `<<:` and `&` make things more complex when we have filed replacement scenario. It looks like in this case, CRD generation is still quite useful when we have kind of complex differences between versions. @Harwayne What do you think of it?
> 
> I slightly checked his PR. I think for channel CRD, maybe he could put v1beta1 which have less field as the first schema, then insert additional fields that v1alpha1 have as the second schema, maybe this will make the syntax cleaner?

So v1beta1 has a different set of properties. Instead of subscribable (which is there in v1alpha1 in spec.properties), v1beta1 has subscribers (which is a layer beneath subscribable). The same for deadLetterSink (which exists in v1alpha1 in subscribable.properties.subscribers.items.properties, but doesn't in v1beta1). Since the property keys are different, regardless of which one version is described, it will involve copying down the fields in the yaml merge.

**Comment by bharattkukreja** [comment]
@capri-xiyue Integration tests are finally passing. Problem was that subscriberURI and replyURI from v1alpha1 in channel.yaml are changed to subscriberUri and replyUri in v1beta1 (different casing), which I had missed. Anyways, this is ready for review.

**Pull request merged**